src/store/flow/block-types/MobilePrimitives_NumericResponseBlockStore.ts

```diff
@@ -68,21 +69,45 @@ export const actions: ActionTree<IFlowsState, IRootState> = {
         prompt: blankResource.uuid,
         validation_minimum: undefined,
         validation_maximum: undefined,
+        ...await dispatch('initiateExtraVendorConfig'),
       },
       tags: [],
       vendor_metadata: {},
     })
   },
 
+  /**
+   * Override this in the consumer side to add extra config props to avoid the validation saying we have missing prop at the creation
+   * eg: {
+   *   prop1: undefined,
+   *   prop2: undefined,
+   * }
+   */
+  async initiateExtraVendorConfig(_ctx: unknown): Promise<object> {
+    return {}
+  },
+
   handleBranchingTypeChangedToUnified({dispatch}, {block}: {block: IBlock}) {
     dispatch('flow/block_convertExitFormationToUnified', {
       blockId: block.uuid,
       test: 'is_number(block.value)',
     }, {root: true})
   },
 
-  validate({rootGetters}, {block, schemaVersion}: {block: IBlock, schemaVersion: string}) {
-    const validationResults = validateCommunityBlock({block, schemaVersion})
+  /**
+   * Validate the vendor block (Consumer block)
+   * By overriding this action in the consumer side, we will be able to customize it using different json schema for eg.
+   *
+   * Important: This will be overridden in the consumer side, so DO NOT add generic validations here,
+   * instead edit the `validate()` if needed.
+   */
+  async validateVendorBlock({rootGetters}, {block, schemaVersion}: {block: IBlock, schemaVersion: string}): Promise<IValidationStatus> {
+    const blockTypeWithoutNameSpace = block.type.split('.')[block.type.split('.').length - 1]
+    return validateCommunityBlock({block, schemaVersion, customBlockJsonSchema: require(`@floip/flow-runner/dist/resources/validationSchema/${schemaVersion}/I${blockTypeWithoutNameSpace}Block.json`)})
+  },
+
+  async validate({rootGetters, dispatch}, {block, schemaVersion}: {block: IBlock, schemaVersion: string}) {
```

Comment:
I wish the `validate` was called differently, to distinguish from `validateVendorBlock` and `validateCommunityBlock`.
Something like `validateAndUpdateErrorNotifications`(?). But that would mean changing this across all block stores.
Any ideas?

Reply:
I'd keep it this short if you don't have a very strong opinion here. To reduce the confusing, let me add a js doc for that generic validate action. Thanks for bringing this up